test/extensions/filters/http/rbac/rbac_filter_test.cc

```diff
@@ -117,6 +141,15 @@ TEST_F(RoleBasedAccessControlFilterTest, RequestedServerName) {
   Buffer::OwnedImpl data("");
   EXPECT_EQ(Http::FilterDataStatus::Continue, filter_.decodeData(data, false));
   EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_.decodeTrailers(trailers_));
+
+  // Check Log
+  setMetadata();
```

Comment:
Call the setMetadata() in the beginning, also remove the extra call of `EXPECT_EQ(Http::FilterHeadersStatus::Continue, ...)` which is already called above.

Reply:
This checks the log action in the same test as the regular action. After refactoring some stuff from before, it might make sense to do all log tests this way. Do you think its better to combine log tests with Allow tests or should we keep them separate?

Reply:
let's combine them together so that we also cover the case that non-LOG action will not write to the metadata accidentally.

Reply:
I mean, testing that LOG action will log whenever ALLOW action allows. So I'm using two separate configs in these tests. This allows me to easily add checks for log action for other cases like Path and RequestedServerName. Do you think that's too confusing?

Reply:
I think this is unnecessary and complicates the tests with two configs. Like I said before, just initialize the config to the action you want to test, and then check the behavior:
- for ALLOW action, it should allow the request and deny by default, the metadata should not change
- for DENY action, it should deny if explicitly matched and allow if not, the metadata should not change
- for LOG action, it should always allow, and the metadata should change if matched

Reply:
Sure sounds good.